src/server/scripts/Pandaria/GateoftheSettingSun/boss_raigonn.cpp

```diff
@@ -0,0 +1,664 @@
+
+                if (!whoPlayer)
+                    return;
+
+                if (whoPlayer->IsGameMaster())
+                    return;
+
+                inFight = true;
+                DoZoneInCombat();
+                events.CancelEventGroup(PHASE_VULNERABILITY);
+                events.ScheduleEvent(EVENT_SUMMON_PROTECTORAT, urand(15000, 30000));
+                events.ScheduleEvent(EVENT_SUMMON_ENGULFER, urand(15000, 30000));
+                events.ScheduleEvent(EVENT_SUMMON_SWARM_BRINGER, urand(15000, 30000));
+                events.ScheduleEvent(EVENT_CHECK_WIPE, 1000);
+
+                pInstance->SetBossState(DATA_RAIGONN, IN_PROGRESS);
+
+                if (Creature* weakPoint = pInstance->GetCreature(pInstance->GetData64(NPC_WEAK_SPOT)))
+                {
+                    instance->SendEncounterUnit(ENCOUNTER_FRAME_ENGAGE, weakPoint);
+                    weakPoint->setFaction(16);
+                    weakPoint->ClearUnitState(UNIT_STATE_UNATTACKABLE);
+                    weakPoint->RemoveFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_NON_ATTACKABLE);
+                }
+            }
+
+            void EnterCombat(Unit* who)
+            {
+                if (Phase != PHASE_VULNERABILITY)
+                    return;
+
+                _EnterCombat();
+            }
+
+            void MovementInform(uint32 type, uint32 pointId)
+            {
+                if (type != POINT_MOTION_TYPE && type != EFFECT_MOTION_TYPE)
+                    return;
+
+                switch (pointId)
+                {
+                    case POINT_MAIN_DOOR:
+                    case POINT_HERSE:
+                        DoEventCharge();
+                        break;
+                }
+            }
+
+            void DoAction(int32 action)
+            {
+                if (action == ACTION_WEAK_SPOT_DEAD)
+                {
+                    Phase = PHASE_VULNERABILITY;
+                    me->SetReactState(REACT_AGGRESSIVE);
+                    me->SetSpeed(MOVE_RUN, 1.1f);
+
+                    me->CastStop();
+                    me->RemoveAurasDueToSpell(SPELL_IMPERVIOUS_CARAPACE);
+                    me->CastSpell(me, SPELL_BROKEN_CARAPACE, true);
+                    me->CastSpell(me, SPELL_BROKEN_CARAPACE_DAMAGE, true);
+
+                    events.CancelEventGroup(PHASE_WEAK_SPOT);
+                    events.ScheduleEvent(EVENT_FIXATE, 30000, PHASE_VULNERABILITY);
+                    events.ScheduleEvent(EVENT_STOMP, 16000, PHASE_VULNERABILITY);
+
+                    DoZoneInCombat();
+                    if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM))
+                        AttackStart(target);
+                }
+            }
+
+            void JustReachedHome()
+            {
+                instance->SetBossState(DATA_RAIGONN, FAIL);
+                summons.DespawnAll();
+            }
+
+            void JustSummoned(Creature* summoned)
+            {
+                summons.Summon(summoned);

```

Comment:
I think this is already of part of parent class BossAI, should not be needed, check ScriptedCreature.h/cpp